### ISSUE TITLE
Unable to use library on ubunt 22.04 due to libssl1.1 dependancy.

### ISSUE THREAD

**Issue by JasoonS** [LimeChain/matchstick#344]
Ubuntu 22.04 only includes libssl3 (not libssl.1.1)

```
matchstick: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or director
```

Same goes for `libcrypto.so.1.1`.

If I link the libraries(`sudo ln -s /usr/lib/x86_64-linux-gnu/libssl.so.3 /usr/lib/x86_64-linux-gnu/libssl.so.1.1`, `sudo ln -s /usr/lib/x86_64-linux-gnu/libcrypto.so.3 /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1`) I get the following error:
```
../matchstick: /lib/x86_64-linux-gnu/libssl.so.1.1: version `OPENSSL_1_1_0' not found (required by ../matchstick)
../matchstick: /lib/x86_64-linux-gnu/libcrypto.so.1.1: version `OPENSSL_1_1_0' not found (required by ../matchstick)
```

Only solution is to install an officially unsuported version:
```
wget archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2_amd64.deb
sudo dpkg -i ./libssl1.1_1.1.1f-1ubuntu2_amd64.deb
```

**Comment by dimitrovmaksim** [comment]
Currently there's no binary for ubuntu 22 and it's downloading the one for 20 by default

**Comment by dimitrovmaksim** [comment]
matchstick 0.5.0 now has an ubuntu 22.04 binary, but the graph-cli change is not yet merged

**Comment by dimitrovmaksim** [comment]
Support for ubuntu 22.04 should be now available through graph-cli 0.30.3 and matchstick 0.5.0
